aten/src/ATen/native/native_functions.yaml

```diff
@@ -6276,20 +6276,13 @@
 
 - func: qr.Q(Tensor self, bool some=True, *, Tensor(a!) Q, Tensor(b!) R) -> (Tensor(a!) Q, Tensor(b!) R)
   dispatch:
-    DefaultBackend: qr_out
+    Math: qr_out
 
 - func: qr(Tensor self, bool some=True) -> (Tensor Q, Tensor R)
   use_c10_dispatcher: full
   variants: method, function
   dispatch:
-    DefaultBackend: qr
-
-- func: _qr_helper(Tensor self, bool some) -> (Tensor, Tensor)
-  use_c10_dispatcher: full
-  variants: function
-  dispatch:
-    CPU: _qr_helper_cpu
-    CUDA: _qr_helper_cuda
+    Math: qr
```

Comment:
AFAIK This function has a custom dispatch and so is not a composite function. This means that it should have a gradient formula defined in derivatives.yaml otherwise it will be marked as non-differentiable automatically.

Could you check the generated code for ` qr(` in `torch/csrc/autograd/generated/VariableTypeEverything.cpp` please to make sure it is still properly differentiable?

Reply:
> Could you check the generated code for qr( in torch/csrc/autograd/generated/VariableTypeEverything.cpp please to make sure it is still properly differentiable?

I'm not very familiar with that code so I'm not sure what exactly I should check. But there are tests which check that `qr` is differentiable and they all pass.

The trick is that now `torch.qr` and `torch.lingalg.qr` are thin wrappers around `_qr_helper`: so `derivatives.yml` contains just the formula for the latter.

Reply:
Yep; I think autograd is working as expected.

Reply:
ok, marking this thread as resolved then